cylc/flow/broadcast_mgr.py

```diff
@@ -83,7 +83,12 @@ def check_ext_triggers(self, itask, ext_trigger_queue):
         return self._match_ext_trigger(itask)
 
     def clear_broadcast(
-            self, point_strings=None, namespaces=None, cancel_settings=None):
+        self,
+        point_strings=None,
+        namespaces=None,
+        fail_points=None,
```

Comment:
I'm struggling to understand this handling of `fail_points`. Why isn't it handled in `cancel_settings`?

Reply:
Because I honestly don't understand how this all fits together with the calling of the mutation by the CLI, and I'm not sure how to do this. I think `cancel_keys_list` is there for the cli rather than internal interfaces.

Reply:
We shouldn't require any changes to the broadcast code for simulation mode.

Try casting the cycle point in the simulation code.

Reply:
> We shouldn't require any changes to the broadcast code for simulation mode.

Seems reasonable - I've removed it, just one test failure as a result.

> Try casting the cycle point in the simulation code rather than the broadcast code.

Sorry, not sure I understand

> The simulation code should not need to call the clear_broadcast interface?

What should the sim mode code do if the Fail CP is garbage.

Reply:
I've dealt with this after a conversation on teams with @oliver-sanders . :)